Sources/BSON/BSONObjectID.swift

```diff
@@ -46,3 +69,29 @@ extension BSONObjectID: BSONValue {
         buffer.writeBytes(self.oid)
     }
 }
+
+/// A class responsible for generating ObjectIDs for a given instance of this library
+/// An ObjectID consists of a random number for this process, a timestamp, and a counter
+internal class ObjectIDGenerator {
+    private let randomNumber: [UInt8]
+    private var counter: NIOAtomic<UInt32>
```

Comment:
it might be useful to note somewhere, either here or below where it's used that we only use the three lower bytes of this value